lib/queryBuilder/graphUpserter/UpsertNode.js

```diff
@@ -96,6 +97,19 @@ function hasChanges(currentModel, upsertModel, queryProps) {
 
     if (!relations[key] && currentModel[key] !== upsertModel[key]) {
       return true;
+    } else if (relations[key] instanceof BelongsToOneRelation) {
+      const relatedProp = relations[key].relatedProp;
+      const hasChanges = [];
+
+      for (let i = 0, l = relatedProp.size; i < l; ++i) {
```

Comment:
Does this also support the json keys case you commented about earlier?

Reply:
Yes. `relation` abstracts away those differences. If you use `relation.getProp` it will select the correct implementation based on the relation type. If the foreign key is a property in a json (or some other non-scalar) column, `getProp` fetches the nested value correctly.

Reply:
Cool! :) Is there any plans/tickets concerning adding these things to the docs? Or are docs only released on certain version bumps?